doc/introduction/quantum_datasets.rst

```diff
@@ -6,4 +6,206 @@
 Quantum Datasets
 =================
 
-Placeholder
+PennyLane provides the :mod:`~.qdata` module to access quantum datasets comprising of quantum
+chemsitry data and quantum many-body spin systems data. It contains the functionalities to
+download these datasets from a server, store and manipulate it locally. It also provides tools
+for users to build such datasets themselves using the :class:`~.pennylane.qdata.QuantumData` class.
+Here we explain the structure of the hosted data and how can it be used inside PennyLane.
+
+Dataset Structure
+------------------
+
+PennyLane's quantum dataset currently contains two subcategories: `qchem` and `qspin`, which
+contains data regarding molecules and spin systems, respectively. Users can use the 
+:func:`~.pennylane.qdata.list_datasets` method to get a snapshot of the current state of the
+datasets as we show below:
+
+.. code-block:: python
+
+    >>> from pprint import pprint
+    >>> print('Level 1:'); pprint(qdata.list_datasets(), depth=1)
+    Level 1:
+    {'qchem': {...}, 'qspin': {...}}
+    >>> print('Level 2:'); pprint(qdata.list_datasets(), depth=2)
+    Level 2:
+    {'qchem': {'H2': {...}, 'LiH': {...}, 'NH3': {...}, ...},
+     'qspin': {'Heisenberg': {...}, 'Ising': {...}, ...}}
+
+This nested-dictionary structure can also be used to generate arguments for the :func:`~.pennylane.qdata.load`
+function that allows us to downlaod the dataset to be stored and accessed locally. The main purpose of these
+arguments is to proivde users with the flexibility of filtering data as per their needs and downloading what
+matches their specified criteria. For example, 
+
+.. code-block:: python
+
+    >>> qdata.get_params(qdata.list_datasets(), "qchem", basis=["STO3G"])
+    [{'molname': ['full'], 'basis': ['STO3G'], 'bondlength': ['full']}]
+    >>> qdata.get_keys("qchem", qdata.get_params(qdata.list_datasets(), "qchem",)[0])
+    ['full']
+
+These arguments can be supplied as it is with the load function or users can manually built these arguments
+as per their liking. Upon doing so, they can simply load the data as follows:
+
+.. code-block:: python
+
+    >>> data_type = "qchem"
+    >>> data_params = {"molname":"full", "basis":"full", "bondlength":"full"}
+    >>> dataset = qdata.load(data_type, data_params)
+    Downloading data to datasets/qchem
+    [███████████████████████████████████████████████████████ 100.0 %] 146.48 KB/146.48 KB
+    >>> dataset
+    [<pennylane.qdata.chemdata.Chemdata at 0x1666b2c50>,
+     <pennylane.qdata.chemdata.Chemdata at 0x1666b2500>,
+     <pennylane.qdata.chemdata.Chemdata at 0x28917aec0>]
+
+Using Datasets in PennyLane
```

Comment:
I think this section should be brought up. Can we start with a simple example that:

- Loads the molecular Hamiltonian for water and uses it in a circuit.
- The fill-CI energy is accessed to compare the computed `<H>`.

Later, we can provide more details on the data structure and more advanced features.